bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/link/ItemChannelLinkRegistry.java

```diff
@@ -229,4 +270,171 @@ public enum ItemChannelLinkProblem {
         ITEM_MISSING,
         ITEM_AND_THING_CHANNEL_MISSING;
     }
+
+    /**
+     * If the item does not already have a Point and/or a Property tag and if the linked channel has
+     * 'useTags=true' then assign the default tags from that channel to the respective item. By contrast
+     * if the item does already have a Point and/or a Property tag then we write a warning message in
+     * the log. The warning is also logged if the item has more than one linked channel with 'useTags=true'.
+     * And in any case if the item has native custom tags then those tags remain.
+     */
+    private void assignChannelDefaultTags(ItemChannelLink link, ActiveItem activeItem) {
+        boolean alreadyHasPointOrPropertyTag = false;
+
+        for (String tag : activeItem.getTags()) {
+            Class<? extends Tag> type = SemanticTags.getById(tag);
+            if ((type != null) && (Point.class.isAssignableFrom(type) || Property.class.isAssignableFrom(type))) {
+                alreadyHasPointOrPropertyTag = true;
+                break;
+            }
+        }
+
+        Set<String> channelDefaultTags = getChannelDefaultTags(link);
+        if (!channelDefaultTags.isEmpty()) {
+            if (alreadyHasPointOrPropertyTag) {
+                logger.warn("Item '{}' already tagged; forbidden to assign tags from channel '{}'.",
+                        activeItem.getName(), link.getLinkedUID());
```

Comment:
Why a warning?!
This is just a normal case, certainly even the standard case until this new feature is added.
This would generate massive logs for no reason as there is nothing to warn.
In my opinion, there is nothing to log, or just a DEBUG log.

Reply:
It means that the user has in the .items file the user self intentionally applied **both** Point/Property at the native item level and also declared useTags at the Channel level. So we have to warn the user that they cannot have both.

HOWEVER having said that, if we go the way of applying alwaysUseTags at global level then indeed we should probably not log it.

Reply:
> HOWEVER having said that, if we go the way of applying alwaysUseTags at global level then indeed we should probably not log it.

Yes
I assume that most of users will enable this setting if they are informed of its existence, at least of course I will myself.

Reply:
Moi aussi